numpy/core/src/private/ufunc_override.c

```diff
@@ -32,8 +32,8 @@ has_non_default_array_ufunc(PyObject *obj)
                                                      "__array_ufunc__");
     }
 
-    /* Fast return for ndarray */
-    if ((PyObject *)Py_TYPE(obj) == ndarray) {
+    /* Fast return for ndarray and basic types */
+    if ((PyObject *)Py_TYPE(obj) == ndarray || _is_basic_python_type(obj)) {
```

Comment:
yes, it shouldn't be needed here if we'd call GetAttr with the object not its type, but I don't dunerstand why that is done.
Shouldn't the object have the function if the class has it?